File: easy/branch-sums/branch-sums.js
function branchSums(root) {
	let solution = [];
	helper(root, 0, solution)
	return solution;
}

function helper(node, runningSum, solution) {
	if (!node) return;
	const newRunningSum = runningSum + node.value;
	if (!node.left && !node.right) return solution.push(newRunningSum);
	helper(node.left, newRunningSum, solution);
	helper(node.right, newRunningSum, solution);
}

File: easy/branch-sums/branch-sums.py
def branchSums(root):
    # Write your code here.
	arr = []
	helper(root, 0, arr)
	return arr

def helper(node, pathSum, result):
	if node is None:
		return
	
	newRunningSum = pathSum + node.value
	
	if node.right is None and node.left is None:
		result.append(newRunningSum)
		return
	
	helper(node.left, newRunningSum, result)
	helper(node.right, newRunningSum, result)

